windows/msvc: Implement file/directory type query.

stinos opened the pull request:
Add some more POSIX compatibility by adding a d_type field to the
dirent structure and defining corresponding macros so listdir_next
in the unix' port modos.c can use it, end result being uos.ilistdir
now reports the file type.

Note I fixed some whitespace issues; I recently started using clang-format for MicroPython as well and it matches the default syle pretty well (only changes indentation of the license comment) so no more worries about getting all indentation and whatnot right. If anyone is interested, this is the .clang-format file I have now:
```
---
BasedOnStyle: LLVM
IndentWidth: 4
TabWidth: 4
Language: Cpp
PointerAlignment: Right
ColumnLimit: 100
```
Ideally this would be part of the repository and everybody would use it pre-commit.

pfalcon commented:
> Note I fixed some whitespace issues

Common wisdom says that this should be a separate commit; style cleanups should not add a background noise for the functional changes. See e.g. https://github.com/micropython/micropython/wiki/ContributorGuidelines#how-to-contribute (2nd-to-last para).

stinos committed:
windows/msvc: Fix incorrect indentation in dirent.c.

stinos committed:
windows/msvc: Implement file/directory type query.

Add some more POSIX compatibility by adding a d_type field to the
dirent structure and defining corresponding macros so listdir_next
in the unix' port modos.c can use it, end result being uos.ilistdir
now reports the file type.

stinos commented:
Sorry about that, I thought smallish changes were ok now but I'm must be confusing this with another repository. Now fixed.

dpgeorge commented:
Thanks, looks good, merged in 397ee7c00e67ba4bd2da5d93187c47725f55157b and 02ca8d4674773ed6258eb435beaf0004e04a56ac

dpgeorge commented:
> I recently started using clang-format for MicroPython as well and it matches the default syle pretty well

I did try using astyle myself but it forced some things to be different and that made it a deal breaker for astyle.

I did also try clang-format but I didn't find that it matched well at all.  I'll need to try the above settings, and maybe it can be added as a check to PRs.

stinos commented:
Not really a fan of astyle myself, becaause it's purely lexical (I think) there's always some funny edge cases while clang-format on the other hand builds an AST. As far as I can see the above settings only have 2 mismatches wrt the uPy style: I'll see if I can get it to leave the license comment alone and also not reorder includes alphabetically.
